packages/core/src/payment/strategies/amazon-pay-v2/amazon-pay-v2-payment-processor.ts

```diff
@@ -41,6 +43,104 @@ export default class AmazonPayV2PaymentProcessor {
         return Promise.resolve();
     }
 
+    renderAmazonPayButton(
+        containerId: string,
+        checkoutState: InternalCheckoutSelectors,
+        methodId: string,
+        placement: AmazonPayV2Placement,
+        options?: AmazonPayV2ButtonInitializeOptions
+    ): HTMLElement {
+        const container = document.getElementById(containerId);
+
+        if (!container) {
+            throw new InvalidArgumentError('Unable to render the Amazon Pay button to an invalid HTML container element.');
+        }
+
+        const amazonPayV2ButtonOptions = options ?? this._getAmazonPayV2ButtonOptions(checkoutState, methodId, placement);
+
+        this.createButton(`#${containerId}`, amazonPayV2ButtonOptions);
+
+        return container;
+    }
+
+    private _getAmazonPayV2ButtonOptions(
+        {
+            paymentMethods: { getPaymentMethodOrThrow },
+            config: { getStoreConfigOrThrow },
+            cart: { getCart },
+        }: InternalCheckoutSelectors,
+        methodId: string,
+        placement: AmazonPayV2Placement
+    ): AmazonPayV2ButtonInitializeOptions {
+        const {
+            config: {
+                merchantId,
+                testMode,
+            },
+            initializationData: {
+                checkoutLanguage,
+                ledgerCurrency,
+                checkoutSessionMethod,
+                extractAmazonCheckoutSessionId,
+                publicKeyId,
+                createCheckoutSessionConfig,
+            },
+        } = getPaymentMethodOrThrow(methodId);
+
+        const {
+            checkoutSettings: { features },
+            storeProfile: { shopPath },
+        } = getStoreConfigOrThrow();
+
+        const cart = getCart();
+
+        if (!merchantId || !ledgerCurrency) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const buttonBaseConfig = {
+            merchantId,
+            ledgerCurrency,
+            checkoutLanguage,
+            productType: cart && getShippableItemsCount(cart) === 0 ?
+                AmazonPayV2PayOptions.PayOnly :
+                AmazonPayV2PayOptions.PayAndShip,
+            placement,
+            buttonColor: AmazonPayV2ButtonColor.Gold,
+        };
+
+        if (features['PROJECT-3483.amazon_pay_ph4']) {
+            return /^(SANDBOX|LIVE)/.test(publicKeyId)
+                ? {
+                      ...buttonBaseConfig,
+                      publicKeyId,
+                      createCheckoutSessionConfig,
+                  }
+                : {
+                      ...buttonBaseConfig,
+                      sandbox: !!testMode,
+                      createCheckoutSessionConfig: {
+                          ...createCheckoutSessionConfig,
+                          publicKeyId,
+                      },
+                  };
+        }
+
+        const createCheckoutSession = {
```

Comment:
🥇

Reply:
Hi @vmparra, code coverage is not working for me neither local nor here as you can see below on ci/circleci: test artifacts. 👇🏻😬